test/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -138,4 +138,40 @@ using ChainRules: level2partition, level3partition, chol_blocked_rev, chol_unblo
             @test chol_unblocked_rev(B̄, B, true) ≈ chol_blocked_rev(B̄, B, 10, true)
         end
     end
+    @testset "qr" begin
+        @testset "the thing" begin
+            for n in [4, 6, 10], m in [4, 6, 10]
+                X = randn(n, m)
+
+                if n < m
+                    @test_throws ArgumentError F, dX_pullback = rrule(qr, X)
+                    continue
+                end
+
+                F, dX_pullback = rrule(qr, X)
+                for p in [:Q, :R]
+                    Y, dF_pullback = rrule(getproperty, F, p)
```

Comment:
does `rrule_test` not work on this?

Reply:
It does not work on `getproperty` because of the failure of `FiniteDifferences.jl` mentionned elsewhere on this PR.

It does not work directly on `qr` either, because the object returned by `qr` can not be `collect`ed. See https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/74